src/tests/amp/ampEpicTests.ts

```diff
@@ -23,46 +24,88 @@ const fetchAmpEpicTests = (): Promise<AmpEpicTest[]> =>
             return data.tests;
         });
 
-const [, getCachedAmpEpicTests] = cacheAsync<AmpEpicTest[]>(
+export const [, getCachedAmpEpicTests] = cacheAsync<AmpEpicTest[]>(
     fetchAmpEpicTests,
     60,
     'ampEpicTests',
     true,
 );
 
-export const selectAmpEpicTest = async (
+type AmpExperiments = {
+    [key: string]: {
+        variants: {
+            [key: string]: number;
+        };
+    };
+};
+
+export const getAmpExperimentData = async (): Promise<AmpExperiments> => {
+    const cachedTests = await getCachedAmpEpicTests();
+    const ampExperiments: AmpExperiments = {
+        FALLBACK: {
+            variants: {
+                control: 99.9999999999, // 100.0 is not valid for AMP
+            },
+        },
+    };
+
+    cachedTests.forEach((test: AmpEpicTest) => {
+        const variants: { [key: string]: number } = {};
+
+        const variantNames = test.variants.map(variant => variant.name);
+        const variantAudiencePercentage = 100 / variantNames.length;
+
+        variantNames.forEach(variantName => {
+            variants[variantName.toUpperCase()] = variantAudiencePercentage - 0.0000000001;
+        });
+
+        ampExperiments[test.name.toUpperCase()] = {
```

Comment:
is there a reason for uppercasing everything? I'm wondering what this does to the analytics...

Reply:
Although since this applies to all AMP tests and only AMP tests it doesn't really matter!

Reply:
I just did this to ensure case consistency, as the variant names are used as keys on the `ampExperiments` object. It could be all lower case instead if that would be more in-keeping with current style?

Reply:
Elsewhere in this file we just refer to `test.name` and `variant.name`. Wouldn't it be consistent to just always use whatever string comes from the tool?

Reply:
My main concern is that the test/variant names that go into data lake are consistent (for both the view and acquisition events) are consistent